From ad27834d7c265ea3c718323e98ab97848798bcb2 Mon Sep 17 00:00:00 2001
From: fjebaker <fergusbkr@gmail.com>
Date: Mon, 12 Sep 2022 15:08:31 +0100
Subject: [PATCH 1/1] sqlite: patches to work

---
 sqlite.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sqlite.zig b/sqlite.zig
index cd192a2..05d513c 100644
--- a/sqlite.zig
+++ b/sqlite.zig
@@ -332,7 +332,7 @@ pub const Db = struct {
         switch (options.mode) {
             .File => |path| {
                 var db: ?*c.sqlite3 = undefined;
-                const result = c.sqlite3_open_v2(path, &db, flags, null);
+                const result = c.sqlite3_open_v2(@ptrCast([*c]const u8, path), &db, flags, null);
                 if (result != c.SQLITE_OK or db == null) {
                     if (db) |v| {
                         diags.err = getLastDetailedErrorFromDb(v);
-- 
2.32.1 (Apple Git-133)

